Userland/Libraries/LibWeb/HTML/Location.cpp

```diff
@@ -310,10 +310,43 @@ WebIDL::ExceptionOr<String> Location::search() const
     return TRY_OR_THROW_OOM(vm, String::formatted("?{}", url.query()));
 }
 
-WebIDL::ExceptionOr<void> Location::set_search(String const&)
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#dom-location-search
+WebIDL::ExceptionOr<void> Location::set_search(String const& value)
 {
-    auto& vm = this->vm();
-    return vm.throw_completion<JS::InternalError>(JS::ErrorType::NotImplemented, "Location.search setter");
+    // The search setter steps are:
+    auto const relevant_document = this->relevant_document();
+
+    // 1. If this's relevant Document is null, then return.
+    if (!relevant_document)
+        return {};
+
+    // 2. If this's relevant Document's origin is not same origin-domain with the entry settings object's origin, then throw a "SecurityError" DOMException.
+    if (!relevant_document->origin().is_same_origin_domain(entry_settings_object().origin()))
+        return WebIDL::SecurityError::create(realm(), "Location's relevant document is not same origin-domain with the entry settings object's origin"_fly_string);
+
+    // 3. Let copyURL be a copy of this's url.
+    auto copy_url = this->url();
+
+    if (value.is_empty()) {
+        // 4. If the given value is the empty string, set copyURL's query to null.
+        copy_url.set_query({});
+    } else {
+        // 5. Otherwise, run these substeps:
+
+        // 5.1. Let input be the given value with a single leading "?" removed, if any.
+        auto input = value.bytes_as_string_view().trim("?"sv, TrimMode::Left);
```

Comment:
I dont believe this is to spec as this will trim more than a single trailing '?'.

Reply:
Good catch! Changes were made in https://github.com/LadybirdBrowser/ladybird/pull/1140/commits/f02382a35257c4fd5c2d7f577fd445ba9448e768.
Thank you!